### PR TITLE
Migrate to GitHub Actions, use pyproject.toml for tool config

### PR DESCRIPTION
**Pull request by bittner**
Travis CI has stopped working for a while. We move to GitHub Actions just like many other projects.

We also move tool configuration settings to [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/), which seems to be getting more and more popular among Python tool developers.

Closes #121

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Version Support
 ---------------
 
 *behave-django* is `tested against`_ the officially supported combinations of
-Python and Django (Django 2.2, 3.1, 3.2 on Python 3.6, 3.7, 3.8, 3.9, 3.10).
+Python and Django (Django 2.2, 3.2, 4.0 on Python 3.6, 3.7, 3.8, 3.9, 3.10).
```
Comment (mixxorz) [inline README.rst:28]: I noticed that the badges are still pointing to Travis. Could we change this to show the build status on GitHub Actions?

**Review by mixxorz — APPROVED** [review]
This looks good to me! Thanks for putting in the work to swap to GitHub actions. Also for configuring black!

File: README.rst (hunk 1 above)
Comment (bittner) [reply README.rst:28]: The badges depend on the released version. We would have to publish a new release on PyPI. We could.

**Comment by bittner** [comment]
There are still a few issues:

1. With Travis CI we also ran the test against [the `behave-latest` target](https://github.com/behave/behave-django/blob/1.4.0/.travis.yml#L34). That needs to be added to GHA.
2. The test matrix spun up by the GHA configuration is the full combination of all Python and Django versions, but some combinations are not supported and hence excluded in the Tox configuration (e.g. Django 4.0 doesn't run on Python 3.6, 3.7, Django 2.2 not on Python 3.10). The related jobs need to be excluded from the matrix.

**Comment by bittner** [comment]
This look complete now. I'm merging.

**Pull request merged**